File: packages/silo-client-java/src/main/java/in/org/quicko/silo/client/scope/Project.java
package in.org.quicko.silo.client.scope;

/**
 * A project as the listing and create routes answer it. {@code id} is the
 * record's ULID and never changes; {@code name} is what every path addresses,
 * and can be renamed.
 */
public record Project(String id, String name) {}

File: packages/silo-client-dart/lib/src/scope/project.dart
/// A project record. [id] never changes; [name] is what paths address.
final class Project {
  const Project(this.id, this.name);

  factory Project.fromJson(Map<String, Object?> json) => Project(json['id'] as String, json['name'] as String);

  final String id;
  final String name;
}
